samples/basic/button/src/main.c

```diff
@@ -13,89 +14,143 @@
 
 #define SLEEP_TIME_MS	1
 
-#if DT_PHA_HAS_CELL(DT_ALIAS(sw0), gpios, flags)
-#define SW0_FLAGS DT_GPIO_FLAGS(DT_ALIAS(sw0), gpios)
-#else
-#define SW0_FLAGS 0
-#endif
+/*
+ * Devicetree helper macro which gets the 'flags' cell from a 'gpios'
+ * property, or returns 0 if the property has no 'flags' cell.
+ */
+
+#define FLAGS_OR_ZERO(node)						\
+	COND_CODE_1(DT_PHA_HAS_CELL(node, gpios, flags),		\
+		    (DT_GPIO_FLAGS(node, gpios)),			\
+		    (0))
+
+/*
+ * Get button configuration from the devicetree sw0 alias.
+ *
+ * At least a GPIO device and pin number must be provided. The 'flags'
+ * cell is optional.
+ */
 
-#if DT_PHA_HAS_CELL(DT_ALIAS(led0), gpios, flags)
-#define LED0_FLAGS DT_GPIO_FLAGS(DT_ALIAS(led0), gpios)
+#define SW0_NODE	DT_ALIAS(sw0)
+
+#if DT_HAS_NODE_STATUS_OKAY(SW0_NODE)
+#define SW0_GPIO_LABEL	DT_GPIO_LABEL(SW0_NODE, gpios)
+#define SW0_GPIO_PIN	DT_GPIO_PIN(SW0_NODE, gpios)
+#define SW0_GPIO_FLAGS	(GPIO_INPUT | FLAGS_OR_ZERO(SW0_NODE))
 #else
-#define LED0_FLAGS 0
+#error "Unsupported board: sw0 devicetree alias is not defined"
+#define SW0_GPIO_LABEL	""
```

Comment:
At this point the compiler will error out right? So is it worth defining those?

Reply:
In response to this and a couple of other comments: yes, it is worth it IMO.

If you do it this way, you get exactly one `#error` in the compiler output which is the relevant message about what is missing from the board, instead of that (very useful for beginners IMO) error at the top of the compiler error output, followed by  undefined variable error messages in the below code that uses these macros.

Since this is a 'basic' sample, I am taking extra lengths to keep the compiler output cleaner and shorter when the board is unsupported to avoid errors like https://github.com/zephyrproject-rtos/zephyr/issues/23169.